src/main/tut/kleisli-categories.md

```diff
@@ -0,0 +1,71 @@
+```Haskell
+type Writer a = (a, String)
+```
+```tut:silent
+type Writer[A] = (A, String)
+```
+................
+```Haskell
+a -> Writer b
+```
+```scala
+A => Writer[B]
+```
+................
+```Haskell
+(>=>) :: (a -> Writer b) -> (b -> Writer c) -> (a -> Writer c)
+```
+```scala
+def >=>[A, B, C](m1: A => Writer[B], m2: B => Writer[C]): A => Writer[C]
+```
+................
+```Haskell
+m1 >=> m2 = \x -> 
+    let (y, s1) = m1 x
+        (z, s2) = m2 y
+    in (z, s1 ++ s2)
+```
+```tut:silent
+object kleisli {
+  implicit class Fishy[A, B](m1: A => Writer[B]) {
+   def >=>[C](m2: B => Writer[C]): A => Writer[C] =
+    x => {
+      val (y, s1) = this.m1(x)
```

Comment:
do you actually need to add `this` here?

Reply:
oh nice catch, it's definitely not needed. just a js habit/wart slipping into the code here